js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,137 @@
+    
+    // don't need to test the constructor specifically, function initDate() {
+    
+    // tests for function setDate(date) {
+    describe('setDate', function () {
+        it("should return a date", function () {
+        
+            // using Chai's expect to assert if this is true/false
+            expect(enhancedDate.setDate()).to.be.an.instanceOf(Date);
+        });
+
+        it("should return a date that is passed in", function () {
+        
+            // using Chai's expect to assert if this is true/false
+            // remember: Date (year, month # starting with Jan == 0, day)
+            expect(enhancedDate.setDate(new Date(2015, 6, 2))).to.be.deep.equal(new Date(2015, 6, 2));
+        });
+    });
+        
+    //	function getDate(returnObj) {
+    describe('getDate', function () {
+                
+        it("should return a number if nothing is passed in", function () {
+            expect(enhancedDate.getDate()).to.be.a('number');
+        });
+        
+        it("should return a Date object if true is passed in", function () {
+            expect(enhancedDate.getDate(true)).to.be.an.instanceOf(Date);
+        });
+    });
+    
+    
+    //	function getDayName() {
+    describe('getDayName', function () {
+        
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day), so this is May 31, 2015
+            enhancedDate.setDate(new Date(2015, 4, 31));
+        });
+        
+        it("should return a string", function () {
+            expect(enhancedDate.getDayName()).to.be.a('string');
+        });
+        
+        it("should return 'Sunday'", function () {
+            expect(enhancedDate.getDayName()).to.equal('Sunday');
+        });
+    });
+        
+    //	function getMonthName() {
+    describe("getMonthName", function () {
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 2015
+            enhancedDate.setDate(new Date(2015, 5, 2));
+        });
+        
+        it("should return a string", function () {
+            expect(enhancedDate.getMonthName()).to.be.a('string');
+        });
+        
+        it("should return 'June'", function () {
+            expect(enhancedDate.getMonthName()).to.equal('June');
+        });
+        
+
+    });
+    
+    //	function isToday() {
+    describe("isToday", function () {
+            
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 2015
+            // set a past date so it will never equal today
+            enhancedDate.setDate(new Date(2015, 5, 2));
+        });
+        
+        it("should return a boolean", function () {
+            expect(enhancedDate.isToday()).to.be.a('boolean');
+        });
+        
+        it("should return not equals today", function () {
+            expect(enhancedDate.isToday()).to.not.equal(new Date());
```

Comment:
Since isToday return a boolean, test the return value by checking against true or false - in this case false: `expect(enhancedDate.isToday()).to.be.false;`